solutions/usaco-1107.mdx

```diff
@@ -66,5 +66,83 @@ print(abs(ans))
 ```
 
 </PySection>
+	
+<JavaSection>
+	
+```java
+import java.io.*;
+import java.util.*;
+
+public class YearOfTheCow {
+	static String[] animals = {"Ox", "Tiger", "Rabbit", "Dragon", "Snake", 
+	"Horse", "Goat", "Monkey", "Rooster", "Dog", "Pig", "Rat"};
+
+	static Map<String, Integer> when_born = new TreeMap<>();
+
+	// given a year, return what animal the year is
+	static String getAnimal(int year)
+	{
+		int a = 0, y = 2021;
+		while (y < year) {
+			y++;
+			a++;
+			if (a == 12) {  // reset once you reach 12
+				a = 0;
+			}
+		}
+		while (y > year) {
+			y--;
+			a--;
+			if (a == -1) {
+				a = 11;
+			}
+		}
+		return animals[a];
+	}
+
+	public static void main (String[] args) {
+		Kattio io = new Kattio();
+		int n;
+		n = io.nextInt();
+		when_born.put("Bessie", 2021);
+
+		String cowa, born, in, relation, animal, year, from, cowb;
+		for (int i = 0; i < n; i++) {
+			cowa = io.next();
+			born = io.next();
+			in = io.next();
+			relation = io.next();
+			animal = io.next();
+			year = io.next();
+			from = io.next();
+			cowb = io.next();
```

Comment:
imo this looks pretty messy, maybe use an array to represent this kind of stuff? It's fine as it is now but a little too messy for my taste. Totally up to you though.

Reply:
most of it is junk though